core/src/iam/verify.rs

```diff
@@ -164,41 +165,72 @@ pub async fn token(kvs: &Datastore, session: &mut Session, token: &str) -> Resul
 			// Create a new readonly transaction
 			let mut tx = kvs.transaction(Read, Optimistic).await?;
 			// Parse the record id
-			let id = syn::thing(&id)?;
+			let mut rid = syn::thing(&id)?;
```

Comment:
Now that we have the `AUTHENTICATE` claim, this logic can be executed outside of the match arm that requires `id` to be set. Otherwise, only tokens that already have an `id` claim will be able to evaluate the `AUTHENTICATE` clause, requiring users to set a dummy claim which I think is unnecessary.

Reply:
Done in [fdb3920](https://github.com/surrealdb/surrealdb/pull/4161/commits/fdb3920483acf58fced4ea16d5c8699b355342bd).